### PR TITLE
Adding logic to remove any duplicate write index targets for an alias

### PR DESCRIPTION
**Pull request by ewolinetz**
### Description
Addresses issue seen in one of the must gathers for https://bugzilla.redhat.com/show_bug.cgi?id=1922943 where somehow Elasticsearch has a write-index that points to two different indices (this is not allowed under normal conditions -- e.g. you cannot mark a second index as the write-index while another one is currently the write-index).

Note: it is still undetermined if this is the reason for all the other issues seen on the bz mentioned above so for now it is not considered a fix for it. Looking to backport to 4.6 for stability reasons.

This will mark all but the latest write-index to not be the write-index and then return the single write-index.

/cc @periklis @blockloop @jcantrill 

/cherry-pick release-5.0
/cherry-pick release-4.6

### CODE REVIEW

File: internal/indexmanagement/scripts.go [hunk 1]
```diff
@@ -254,27 +293,14 @@ exit 1
...
-  echo "Received an empty response from elasticsearch -- server may not be ready"
-  exit 1
-fi
+echo "" > $ERRORS
 
-jsonResponse="$(echo [$writeIndices] | sed 's/}{/},{/g')"
+echo "Index management delete process starting"
 
-if ! writeIndex="$(python /tmp/scripts/getWriteIndex.py "${POLICY_MAPPING}-write" "$jsonResponse")" ; then
+if ! writeIndex="$(getWriteIndex "${POLICY_MAPPING}-write")" ; then
```
Comment (periklis) [inline internal/indexmanagement/scripts.go:303]: Can we spell out the `echo $writeIndex` as a proper exit message to understand what failed where?

File: internal/indexmanagement/scripts.go [hunk 2]
```diff
@@ -207,18 +256,8 @@ fi
...
-  -H"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
-  -HContent-Type:application/json \
-  --retry 5 \
-  --retry-delay 5)
-
-jsonResponse="$(echo [$writeIndices] | sed 's/}{/},{/g')"
-
-if ! writeIndex="$(python /tmp/scripts/getWriteIndex.py "${POLICY_MAPPING}-write" "$jsonResponse")" ; then
+## if true, ensure write-alias points to next index
+if ! writeIndex="$(getWriteIndex "${POLICY_MAPPING}-write")" ; then
```
Comment (periklis) [inline internal/indexmanagement/scripts.go:260]: Can we spell out the `echo $writeIndex` as a proper exit message to understand what failed where?

File: internal/indexmanagement/scripts.go [hunk 3]
```diff
@@ -122,31 +123,79 @@ for i in range(0, len(indices), 25):
...
-
-if [ -z "$writeIndices" ]; then
-  echo "Received an empty response from elasticsearch -- server may not be ready"
-  exit 1
-fi
-
-jsonResponse="$(echo [$writeIndices] | sed 's/}{/},{/g')"
-
-if ! writeIndex="$(python /tmp/scripts/getWriteIndex.py "${POLICY_MAPPING}-write" "$jsonResponse")" ; then
+if ! writeIndex="$(getWriteIndex "${POLICY_MAPPING}-write")" ; then
```
Comment (periklis) [inline internal/indexmanagement/scripts.go:198]: Can we spell out the `echo $writeIndex` as a proper exit message to understand what failed where?

File: internal/indexmanagement/scripts.go [hunk 4]
```diff
@@ -122,31 +123,79 @@ for i in range(0, len(indices), 25):
...
+    --retry-delay 5)
+
+  if [ -z "$aliasResponse" ]; then
+    echo "Received an empty response from elasticsearch -- server may not be ready"
+    exit 1
+  fi
+
+  jsonResponse="$(echo [$aliasResponse] | sed 's/}{/},{/g')"
+
+  if ! writeIndices="$(python /tmp/scripts/getWriteIndex.py "${policy}" "$jsonResponse")" ; then
```
Comment (periklis) [inline internal/indexmanagement/scripts.go:150]: Can we spell out the `echo $writeIndices` as a proper exit message to understand what failed where?

**Review by periklis — CHANGES_REQUESTED** [review]

File: internal/indexmanagement/scripts.go (hunk 1 above)
Comment (ewolinetz) [reply internal/indexmanagement/scripts.go:303]: The `getWriteIndex` function should do that, no?

File: internal/indexmanagement/scripts.go (hunk 4 above)
Comment (ewolinetz) [reply internal/indexmanagement/scripts.go:150]: the python script should do this for us already

**Review by periklis — APPROVED** [review]
/lgtm

**Pull request merged**